api/src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java

```diff
@@ -1,61 +1,115 @@
-    private final ComponentRepository componentRepository;
-    private final CollectorRepository collectorRepository;
-
-    @Autowired
-    public CodeQualityServiceImpl(CodeQualityRepository codeQualityRepository,
-                                  ComponentRepository componentRepository,
-                                  CollectorRepository collectorRepository) {
-        this.codeQualityRepository = codeQualityRepository;
-        this.componentRepository = componentRepository;
-        this.collectorRepository = collectorRepository;
-    }
-
-    @Override
-    public DataResponse<Iterable<CodeQuality>> search(CodeQualityRequest request) {
-        Component component = componentRepository.findOne(request.getComponentId());
-        CollectorItem item = component.getCollectorItems().get(CollectorType.CodeQuality).get(0);
-
-        QCodeQuality quality = new QCodeQuality("quality");
-        BooleanBuilder builder = new BooleanBuilder();
-
-        builder.and(quality.collectorItemId.eq(item.getId()));
-
-        if (request.getNumberOfDays() != null) {
-            long endTimeTarget = new LocalDate().minusDays(request.getNumberOfDays()).toDate().getTime();
-            builder.and(quality.timestamp.goe(endTimeTarget));
-        } else {
-            if (request.validDateRange()) {
-                builder.and(quality.timestamp.between(request.getDateBegins(), request.getDateEnds()));
-            }
-        }
-
-        Iterable<CodeQuality> result;
-        if (request.getMax() == null) {
-            result = codeQualityRepository.findAll(builder.getValue(), quality.timestamp.desc());
-        } else {
-            PageRequest pageRequest = new PageRequest(0, request.getMax(), Sort.Direction.DESC, "timestamp");
-            result = codeQualityRepository.findAll(builder.getValue(), pageRequest).getContent();
-        }
-
-        Collector collector = collectorRepository.findOne(item.getCollectorId());
-        return new DataResponse<>(result, collector.getLastExecuted());
-    }
+	private final CodeQualityRepository codeQualityRepository;
+	private final ComponentRepository componentRepository;
+	private final CollectorRepository collectorRepository;
+
+	@Autowired
+	public CodeQualityServiceImpl(CodeQualityRepository codeQualityRepository,
+			ComponentRepository componentRepository,
+			CollectorRepository collectorRepository) {
+		this.codeQualityRepository = codeQualityRepository;
+		this.componentRepository = componentRepository;
+		this.collectorRepository = collectorRepository;
+	}
+
+	@Override
+	public DataResponse<Iterable<CodeQuality>> search(CodeQualityRequest request) {
+		Component component = componentRepository.findOne(request
+				.getComponentId());
+		CollectorItem item = null;
+		Iterable<CodeQuality> result = null;
+		if (request.getType() == null) {
+			item = component.getCollectorItems().get(CollectorType.CodeQuality)
+					.get(0);
+		} else {
+			switch (request.getType()) {
```

Comment:
could optimize this and use an enum or minimize the duplicate code for better readability ..